staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -767,37 +769,31 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 		client = http.DefaultClient
 	}
 
+	if r.timeout > 0 {
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, r.timeout)
```

Comment:
currently we throttle before we set up the context timeout. this flips that so a rate limited client can now timeout because of throttling before it even makes a request

Reply:
Ah ya, good point. Will fix this in #87599.

Reply:
Fixed.